### PR TITLE
Pipeline, Kristina Chekmareva - 21FPL1

### PR DESCRIPTION
**Pull request by kschekmareva**

### CODE REVIEW

File: lab_6_pipeline/pipeline.py [hunk 1]
```diff
@@ -44,37 +92,60 @@ def __init__(self, lemma: str = "", pos: str = "", tags: str = ""):
...
+        self.tags = tags
 
 
 class ConlluToken:
     """
     Representation of the CONLL-U Token
     """
 
-    def __init__(self, text: str):
+    def __init__(self, text: str, morphological_parameters: MorphologicalTokenDTO, position: int):
```
Comment (dmitry-uraev) [inline lab_6_pipeline/pipeline.py]: please, do not modify methods declarations
if you want to set morphological parameters may be you have other ways to do it?

Comment (dmitry-uraev) [reply lab_6_pipeline/pipeline.py]: inspect set_morphological_parameters method more closely

File: lab_6_pipeline/pipeline.py [hunk 2]
```diff
@@ -44,37 +92,60 @@ def __init__(self, lemma: str = "", pos: str = "", tags: str = ""):
...
+        lemma = self._morphological_parameters.lemma
+        pos = self._morphological_parameters.pos
+        xpos = '_'
+        feats = '_'
+        if include_morphological_tags:
+            feats = self._morphological_parameters.tags if self._morphological_parameters.tags else '_'
+        head = '0'
+        deprel = 'root'
+        deps = '_'
+        misc = '_'
```
Comment (dmitry-uraev) [inline lab_6_pipeline/pipeline.py]: can you avoid creating some variables and return strings directly in return statement?

**Review by dmitry-uraev — CHANGES_REQUESTED** [review]
good start
keep going

**Comment by dmitry-uraev** [comment]
Please, do not forget to respond to my comments

File: lab_6_pipeline/pipeline.py (hunk 1 above)
Comment (kschekmareva) [reply lab_6_pipeline/pipeline.py]: Done. I've changed my constructor. The main modification is to convert attributes with default values. Now, the constructor contains only one protected parameter _text and other attributes have default value. Before this, I've created constructor with explicit declaration of parameters inside the constructor.

File: lab_6_pipeline/pipeline.py (hunk 2 above)
Comment (kschekmareva) [reply lab_6_pipeline/pipeline.py]: maybe yes, I just can return '\t'.join([
        str(self._position),
        self._text,
        self._morphological_parameters.lemma,
        self._morphological_parameters.pos,
        '_',
        feats,
        '0',
        'root',
        '_',
        '_'
    ]) and remove all these variables from this method.
However, I think it is easier for me to set these variables in that way, which was written before.

Comment (dmitry-uraev) [reply lab_6_pipeline/pipeline.py]: Why cant you remove these?

Think about variables creation process from the memory consumption point of view

**Comment by dmitry-uraev** [comment]
Waiting for green PR and we are done

File: lab_6_pipeline/pipeline.py (hunk 2 above)
Comment (kschekmareva) [reply lab_6_pipeline/pipeline.py]: I've changed this. Done.

**Pull request closed without merging**